platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,812 @@
+```java
+   byte[] message = Requirements.messageToSign();
+   List<TssShareSignature> signatures = service.sign(privateShares, message);
+
+```
+
+######  Aggregate Signatures
+
+```java
+static {
+    List<TssShareSignature> signatures = Requirements.loadSignatures();
+
+    //Validation of individual signatures
+    List<TssShareSignature> validSignatures = new ArrayList<>(signatures.size());
+    for (TssShareSignature s : signatures) {
+        if (service.verify(participantDirectory, publicShares, s)) {
+            validSignatures.add(s);
+        }
+    }
+
+    //Producing an aggregate signature
+    PairingSignature aggregateSignature = service.aggregate(participantDirectory, validSignatures);
+}
+```
+
+#### Swirlds Native Support
+##### Overview
+It helps other modules to load binary libraries packaged inside the library's jar into memory.
+It provides a way to describe the library for different architectures/ OS.
+
+##### Constraints
+This module will not depend on hedera-services artifacts, so it cannot include logging, metrics, configuration, or any other helper module from that repo.
+
+##### Public API
+###### `OperatingSystem`
+**Description**: An enum class listing all supported OS.
+
+**Link**: [OperatingSystem.java](native%2FOperatingSystem.java)
+
+###### `Architecture`
+**Description**: An enum class listing all supported architectures.
+
+**Link**: [Architecture.java](native%2FArchitecture.java)
+
+###### `LibraryDescriptionEntry`
+**Description**: Given that the compilation of binary libraries produces files with different names under different OS and architectures, we need a way to assemble a catalog for all possible forms our library will take.
+A record of 3 elements that defines the name of the binary file of the library to load in a specific system architecture and OS.
+
+###### `LibraryDescription`
+**Description**: A description of the library in all possible systems.
+
+###### `LibraryLoader`
+**Description**: Helper class that will load a library for the correct system
+
+##### Example
+
+```java
+static {
+    List<LibraryDescriptionEntry> entries = Lists.of(
+            new LibraryDescriptionEntry(OperatingSystem.DARWIN, Architecture.ARC64, "libhedera_bls_jni.dylib"),
+            new LibraryDescriptionEntry(OperatingSystem.LINUX, Architecture.AMD64, "libhedera_bls_jni.so") /*,... Others*/);
+    LibraryDescription libHederaDescription = new LibraryDescription(entries);
+    try {
+        LibraryLoader.load(libHederaDescription);
+    } catch (IOException e){
+        //Handle library loading problems
+    }
+}
+```
+
+##### Implementation details
+Loading a library for a system architecture where no description is provided will result in a runtime exception.
+
+Under the hood, LibraryLoader will make use of:
+https://github.com/hashgraph/full-stack-testing/blob/c3fd5602525145be132770116f5bb5a1a1922dea/fullstack-core/fullstack-base-api/src/main/java/com/hedera/fullstack/base/api/resource/ResourceLoader.java
```

Comment:
This is expressly a test file, so perhaps we mean to say "a class similar to" this will be used?

Reply:
indeed.